model/metrics/clr-metrics.yaml

```diff
@@ -0,0 +1,166 @@
+groups:
+  - id: metric.clr.gc.collections.count
+    type: metric
+    metric_name: clr.gc.collections.count
+    brief: "Number of garbage collections that have occurred since the process started."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.objects.size
+    type: metric
+    metric_name: clr.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.allocations.size
+    type: metric
+    metric_name: clr.gc.allocations.size
+    brief: >
+      The number of bytes allocated on the managed GC heap since the process
+      started. The returned value does not include any native allocations.
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.committed_memory.size
+    type: metric
+    metric_name: clr.gc.committed_memory.size
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
+      observed during the latest garbage collection.
+    note: >
+      Committed virtual memory may be larger than the heap size because it
+      includes both memory for storing existing objects (the heap size) and some
+      extra memory that is ready to handle newly allocated objects in the
+      future.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.heap.size
+    type: metric
+    metric_name: clr.gc.heap.size
+    brief: >
+      The heap size (including fragmentation), as observed during the latest
+      garbage collection.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.heap.fragmentation.size
+    type: metric
+    metric_name: clr.gc.heap.fragmentation.size
+    brief: >
+      The heap fragmentation, as observed during the latest garbage collection.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.duration
+    type: metric
+    metric_name: clr.gc.duration
+    brief: "The total amount of time paused in GC since the process started."
+    instrument: counter
+    unit: "ns"
```

Comment:
Similarly [from the recommendations](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-units): 

> When instruments are measuring durations, seconds (i.e. s) SHOULD be used.

And here, for `jvm.gc.duration` [Java uses seconds as well](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmgcduration).

Since this is GC time paused since process start, I think following Java and the recommendation makes sense - I think this metric will be relevant when the process was running already fairly long and the GC pauses are long as well, so safely this will be mostly in the seconds+ interval.

Reply:
Ah, okay. We can make the change. I am worried that the lack of granularity may make this metric less useful since pauses should be short (sub 1 second) in almost all cases. I see that JVM uses a histogram for this metric with explicit bucket boundaries. The existing contrib instrumentation uses the counter. I'll update to align for now and we can discuss with the runtime team if they think this is a suitable way to measure this for .NET.

Reply:
Actually, having said that, as this is just an incrementing count of the pause time, perhaps using counter is fine here.

Reply:
I assume when using seconds we would be using a floating point value so we can still measure 1ns, it would just be reported as 0.000000001 seconds. That sounds fine to me.